### PR TITLE
RNN Initial Version

### PR DESCRIPTION
**Pull request by hnipun**
@vpj may not be useful. Let me know your thoughts

### CODE REVIEW

File: lab/util/pytorch.py [hunk 1]
```diff
@@ -28,3 +33,52 @@ def get_modules(configs: Configs):
             modules[k] = value
 
     return modules
+
+
+class BaseDataset(Dataset):
+    pass
+
+
+class CustomDataset(BaseDataset):
```
Comment (hnipun) [inline lab/util/pytorch.py:42]: this may not be super general let me know ur thoughts

File: lab/util/pytorch.py [hunk 2]
```diff
@@ -28,3 +33,52 @@ def get_modules(configs: Configs):
...
+        self.test_data = self.data.iloc[self.train_size:]
+
+    @classmethod
+    def from_csv(cls, file_path: str, train: bool, transform: Callable, test_size: float, y_cols: List, x_cols: List):
+        data = pd.read_csv(file_path)
+
+        kwargs = vars()
+        kwargs['data'] = data
+
+        return CustomDataset(**kwargs)
```
Comment (vpj) [inline lab/util/pytorch.py:68]: I think you can just do `CustomDataset(data=data)`

File: lab/util/pytorch.py (hunk 1 above)
Comment (vpj) [reply lab/util/pytorch.py:42]: Shall we move this out from `util/pytorch` and put in in `util/data/pytorch`?

File: lab/util/pytorch.py [hunk 3]
```diff
@@ -28,3 +33,52 @@ def get_modules(configs: Configs):
...
+
+class CustomDataset(BaseDataset):
+    data: pd
+    y_cols: List
+    x_cols: List
+    transform: Callable
+    test_size: float = 0.0
+    train: bool
+
+    def __init__(self, **kwargs):
```
Comment (vpj) [inline lab/util/pytorch.py:50]: Is this initializer being called from anywhere other than `from_csv`?

File: lab/util/pytorch.py (hunk 1 above)
Comment (vpj) [reply lab/util/pytorch.py:42]: Why do you name it `CustomDataset`  shouldn't it be like `PandasDataset` or `CsvDataset`

File: lab/util/pytorch.py [hunk 4]
```diff
@@ -28,3 +33,52 @@ def get_modules(configs: Configs):
...
+class BaseDataset(Dataset):
+    pass
+
+
+class CustomDataset(BaseDataset):
+    data: pd
+    y_cols: List
+    x_cols: List
+    transform: Callable
+    test_size: float = 0.0
```
Comment (vpj) [inline lab/util/pytorch.py:47]: should `test_size` be an `int`?

Comment (vpj) [reply lab/util/pytorch.py:47]: got it, it's a percentage. Can we use a separate name like `test_fraction` for this and keep `test_size` separate

**Pull request merged**